### PR TITLE
Add admin alerts

### PR DESCRIPTION
**Pull request by Chief-Engineer**
## About the PR

Notifies admins of important events such as AME sabotage. Uses a separate channel to allow the alerts to be toggled independently of join/leave messages and because I intend for them to be made more distinct in the future.

`.Owner` is used in the AME alert, please don't make me have to fix the AME stuff to avoid using it.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Server/Administration/Systems/AdminSystem.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ public sealed class AdminSystem : EntitySystem
         [Dependency] private readonly IPlayerManager _playerManager = default!;
         [Dependency] private readonly IAdminManager _adminManager = default!;
 
-        private readonly Dictionary<NetUserId, PlayerInfo> _playerList = new();
```
Comment (metalgearsloth) [inline Content.Server/Administration/Systems/AdminSystem.cs:24]: I think keeping it private is better and adding a method to get the relevant info.

File: Content.Server/AME/Components/AMEControllerComponent.cs [hunk 2]
```diff
@@ -183,6 +186,18 @@ private void OnUiReceiveMessage(ServerBoundUserInterfaceMessage obj)
...
+                if (mindComponent.HasMind)
+                {
+                    var adminSystem = _entities.System<AdminSystem>();
+                    var antag = mindComponent.Mind!.UserId != null
+                                && (adminSystem.GetCachedPlayerInfo(mindComponent.Mind!.UserId.Value)?.Antag ?? false);
+                    if (GetCoreCount() * 2 == InjectionAmount - 2
+                        && msg.Button == UiButton.IncreaseFuel)
+                        _chat.SendAdminAlert(
+                            $"{mindComponent.Mind!.Session?.Name}{(antag ? " (ANTAG)" : "")} increased AME over safe limit to {InjectionAmount}");
+                }
```
Comment (juliangiebel) [inline Content.Server/AME/Components/AMEControllerComponent.cs:200]: Wouldn't it be better to add an overload of the `SendAdminAlert` method that accepts the players uid, the message and an optional mind component which resolves the mind component and prepends the name infront of the message?
```cs
SendAdminAlert(EntityUid player, string message, MindComponent? mindComponent = null)
{
    if(!Resolve(player, ref mindComponent))
    {
        SendAdminAlert(message);
        return;
    }

    var adminSystem = _entities.System<AdminSystem>();
    var antag = mindComponent.Mind!.UserId != null
                                && (adminSystem.GetCachedPlayerInfo(mindComponent.Mind!.UserId.Value)?.Antag ?? false);
    
    SendAdminAlert($"{mindComponent.Mind!.Session?.Name}{(antag ? " (ANTAG)" : "")} {message}");
}
```

**Review by juliangiebel — COMMENTED** [review]
Looks good.
I just have one thing that would make adding more alerts in the future easier because prepending the player name and antag status would probably be needed often imo.

**Review by juliangiebel — APPROVED** [review]
👌

**Comment by mirrorcult** [comment]
im still not really convinced this needs to be separate from the regular admin chat which is already just alerts now, what else will be added to differentiated between them?

**Comment by juliangiebel** [comment]
If the admin chat is just an alert channel now those alerts should use the channel added in this PR so you can filter those imo

**Comment by AJCM-git** [comment]
merge master

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**